### DISCUSSION TITLE
when use store.$reset

### DISCUSSION THREAD

**Discussion by ilaipi** [vuejs/pinia#2311] [Help and Questions] [answered]
### Reproduction

no

### Steps to reproduce the bug

1, define a store, has the state type: 

```
class SearchState {
  q: string;
  current: number;
  pageSize: number;
}
```

2, in a store:

```
state: (): Partial<SearchState> => ({
    current: 1,
    pageSize: 20,
  }),
```

3, in component:

```
const search = () => {
  store.$reset();  // when call $reset, the value of 'q' stay
};
```

### Expected behavior

I think reset should mean that, just back to the init state, all of the fields.

### Actual behavior

only have explicit initial value fields  back to initial value.

**Comment by posva** [comment, marked as answer]
You must define `q` in state for `$reset()` to redefine it.
